### PR TITLE
feat : Operator CRD has OpenAPI spec

### PR DESCRIPTION
**Pull request by bnshr**
Refers [CNFCERT-899](https://issues.redhat.com/browse/CNFCERT-899)

Includes : 
* Check for OpenAPI spec
* Fix for handling larger test result file
* Refactoring of CRD retrieval code and associated new unit tests
* Sorted the expected_results entries.

### CODE REVIEW

**Comment by manurodriguez** [comment]
> from change #1949:
> 
> * FAILURE https://www.distributed-ci.io/jobs/72f23471-c441-4500-ad9f-e6c0f3e5ed77/jobStates

I see you are doing some changes in the OpenAPI spec, and the DCI job failed because that test didn't pass: `{'testcase': 'operator-installed-operator-openapi-schema', 'passed': False}}]"}` 

So I assume this is expected, but please let us know if we need to change anything in our side. Thanks.

File: pkg/autodiscover/autodiscover.go [hunk 1]
```diff
@@ -175,6 +176,14 @@ func DoAutoDiscover(config *configuration.TestConfiguration) DiscoveredTestData
 		os.Exit(1)
 	}
 	data.Crds = FindTestCrdNames(config.CrdFilters)
+	clusterCrds, err := GetClusterCrdNames()
+	if err != nil {
+		if err != nil {
```
Comment (sebrandon1) [inline pkg/autodiscover/autodiscover.go:181]: Two `err` checks?

File: pkg/autodiscover/autodiscover.go [hunk 2]
```diff
@@ -175,6 +176,14 @@ func DoAutoDiscover(config *configuration.TestConfiguration) DiscoveredTestData
 		os.Exit(1)
 	}
 	data.Crds = FindTestCrdNames(config.CrdFilters)
+	clusterCrds, err := GetClusterCrdNames()
```
Comment (sebrandon1) [inline pkg/autodiscover/autodiscover.go:179]: We now have `GetClusterCrdNames()` and `FindTestCrdNames()`.  Would it have been possible to combine the logic?

**Comment by bnshr** [comment]
> > from change #1949:
> > 
> > * FAILURE https://www.distributed-ci.io/jobs/72f23471-c441-4500-ad9f-e6c0f3e5ed77/jobStates
> 
> I see you are doing some changes in the OpenAPI spec, and the [DCI job failed](https://www.distributed-ci.io/jobs/72f23471-c441-4500-ad9f-e6c0f3e5ed77/jobStates?sort=date&task=afddd437-4ecc-4b16-8e95-8506c8f2364b) because that test didn't pass: `{'testcase': 'operator-installed-operator-openapi-schema', 'passed': False}}]"}`
> 
> So I assume this is expected, but please let us know if we need to change anything in our side. Thanks.

I am looking into that - the test shall not fail.

File: pkg/autodiscover/autodiscover.go (hunk 1 above)
Comment (bnshr) [reply pkg/autodiscover/autodiscover.go]: Refactored this part.

File: pkg/autodiscover/autodiscover.go (hunk 2 above)
Comment (bnshr) [reply pkg/autodiscover/autodiscover.go]: Done - the unit tests were a bit tricky though.

**Review by sebrandon1 — APPROVED** [review]

**Pull request merged**